controllers/replica_set_controller.go

```diff
@@ -713,7 +713,32 @@ exec mongod -f /data/automation-mongod.conf ;
 	)
 }
 
+type statefulSetSpec interface {
```

Comment:
I have a few comments/concerns about this

1. I don't think `statefulSetSpec` is a great name for what this is, with this name I would assume it is a type related to `StatefulSet.Spec`. What this actually is is an interface which defines what a resource needs to actually build a StatefulSet.

2. I don't think we should have an unexported interface that is an argument for an exported function.

3. Does it belong in this package? It's currently in the `controllers` package (which I don't think anything should be exported from, this package should just be in charge of the community operator specific controller) I think any functionality that should be shared makes more sense in the `statefulset` package, maybe `statefulset.Owner`/`statefulset.Provider` etc - what do you think?

The reason I can think of against putting it in the statefulset package is that up until now that package has contained functions that acted solely on the statefulset primitives, but I think realistically the package exists to be used between our two projects, and if there was a reason to remove it from there later we could address it then.

4. I think an interface like this needs a docstring.

Reply:
1. Naming is hard :) But I agree with you! I will think about a better name
2. Yeah looks like it's not the best choice by googling a bit, will change it
3. Will try to move things around, I initially tried to have it in another package but was hitting issues with a few circular dependencies, will try again and see what I can get to!
4. Yeah, 100%, thanks for noticing!